src/pal/src/map/virtual.cpp

```diff
@@ -1162,7 +1162,17 @@ VIRTUALCommitMemory(
 
         if (allocationType != MEM_COMMIT)
         {
+
+#if defined(LINUX_OOM_KILL)
+#define LINUX_OOM_KILL_THRESHOLD (32 * 1024)
+            if (PAL_GetMemAvailableFromOS() < LINUX_OOM_KILL_THRESHOLD)
```

Comment:
I don't think this would work. There are many processes running in the system, many of them can be allocating and freeing memory all the time. So it is well possible that between the time you check and the time we try to commit below, other process has used up more memory and the commit would still result in the oom kill. 
Or even other threads in our process can easily cause the same thing. Either by trying to use VirtualCommit or just to allocate memory using malloc.

Reply:
@janvorli Thank you for your opinion. I understand that  generally my approach doesn't provide the solution of this problem. But I don't see other decision besides yet that based on some assessment of the current using memory to issue lack of memory before the oom killer terminates a VM execution